libvast/src/detail/file_path_to_parser.cpp

```diff
@@ -16,11 +16,19 @@ namespace vast::detail {
 
 constexpr inline auto fallback_parser = "json";
 
+const auto filename_to_parser_list
+  = std::array<std::pair<std::string, std::string>, 1>{
+    {{"eve.json", "suricata"}}};
+
 const auto extension_to_parser_list
-  = std::array<std::pair<std::string, std::string>, 2>{
-    {{"eve.json", "suricata"}, {".ndjson", "json"}}};
+  = std::array<std::pair<std::string, std::string>, 1>{{{".ndjson", "json"}}};
 
 auto file_path_to_parser(std::string_view path) -> std::string {
```

Comment:
Semantically, shouldn't this function take a `const std::filesystem::path&`?

Reply:
The `string_view` parameter was a carry-over from the `args` vector, but `const std::filesystem::path&` works better. Changed in https://github.com/tenzir/vast/pull/3129/commits/c190ded73fe29d25a3a41f181966cb72a3ec27fe